.github/workflows/test.yml

```diff
@@ -4,12 +4,12 @@ jobs:
   test:
     strategy:
       matrix:
-        go-version: [1.17.x, 1.18.x]
+        go-version: [1.20.x, 1.21.x]
```

Comment:
Given that this is a library, would you be ok with keeping the "oldest" version a bit conservative? I guess we could either go for oldest + latest;

```yaml
        go-version: [1.17.x, 1.21.x]
```

or "oldest" + "current versions"

```yaml
        go-version: [1.17.x, 1.20.x, 1.21.x]
```


I see the `go.mod` of some modules still says `go1.16`, so perhaps that should even be go1.16, or (with Go now no longer considering it "just an indication")  we may have to update the go.mod in all of those modules.

Reply:
Done (and switched to Go 1.17 everywhere).